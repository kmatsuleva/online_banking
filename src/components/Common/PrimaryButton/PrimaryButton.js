import React from "react";
import PropTypes from 'prop-types';     

const PrimaryButton = ({ value, onClick, btnStyle, ref, disabled}) => (
  <input type = "button" value = {value} onClick = {onClick} style={btnStyle} ref={ref} disabled={disabled}/>
);


PrimaryButton.propTypes = {
  value: PropTypes.string,
  onClick: PropTypes.func,
  btnStyle: PropTypes.object
}

PrimaryButton.defaultProps = {
  btnStyle: {
    backgroundColor: "#46568e",
    boxShadow: "0 1px 2px rgba(0, 0, 0, 0.2)",
    height: 40,
    marginTop: 10,
    borderRadius: 4,
    paddingTop: 8,
    paddingBottom: 8,
    paddingLeft: 20,
    paddingRight: 20,
    fontSize: 16,
    fontFamily: "Roboto, Arial, sans-serif",
    color: "white"
  } 
};


export default PrimaryButton;



