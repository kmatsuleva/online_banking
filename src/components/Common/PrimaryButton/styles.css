.btnStyle {
    background-color: #46568e;
    box-shadow: 0 1px 2px rgba(0, 0, 0, 0.2);
    height: 40px;
    margin-top: 10px;
    border-radius: 4px;
    padding-top: 8px;
    padding-bottom: 8px;
    padding-left: 20px;
    padding-right: 20px;
    font-size: 16px;
    font-family: Roboto, Arial, sans-serif;
    color: white;
}