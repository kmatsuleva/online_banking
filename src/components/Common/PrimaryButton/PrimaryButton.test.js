
import React from 'react'
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import PrimaryButton from "./PrimaryButton"

Enzyme.configure({ adapter: new Adapter() })

describe("< PrimaryButton />", () => {

    it('renders without crashing', () => {
        shallow(<PrimaryButton />);
    });
    
    it('should handle the click event', () => {
        const callback = jest.fn();
        const wrapper = shallow(<PrimaryButton onClick={callback}/>);
        wrapper.simulate('click');
        expect(callback).toHaveBeenCalled();
    });

    it('should render correctly', () => {
        const handleClick = () => {
            alert('clicked');
        };
        const output = shallow(
        <PrimaryButton value="mockValue"
                onClick = {handleClick()}/>
        );
        expect(shallowToJson(output)).toMatchSnapshot();
      });
});