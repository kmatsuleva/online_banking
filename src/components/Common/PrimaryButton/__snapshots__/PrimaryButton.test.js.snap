// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< PrimaryButton /> should render correctly 1`] = `
<input
  style={
    Object {
      "backgroundColor": "#46568e",
      "borderRadius": 4,
      "boxShadow": "0 1px 2px rgba(0, 0, 0, 0.2)",
      "color": "white",
      "fontFamily": "Roboto, Arial, sans-serif",
      "fontSize": 16,
      "height": 40,
      "marginTop": 10,
      "paddingBottom": 8,
      "paddingLeft": 20,
      "paddingRight": 20,
      "paddingTop": 8,
    }
  }
  type="button"
  value="mockValue"
/>
`;
